src/Gui/Clipping.cpp

```diff
@@ -157,37 +185,92 @@ Clipping::Clipping(Gui::View3DInventor* view, QWidget* parent)
             minlen = minlen / steps;
             int dim = static_cast<int>(log10(minlen));
             double singleStep = pow(10.0, dim);
-            d->ui.clipView->setSingleStep(singleStep);
+            Private::ui.clipView->setSingleStep(singleStep);
             minDecimals = std::max(minDecimals, -dim);
         }
         {
             lenx = lenx / steps;
             int dim = static_cast<int>(log10(lenx));
             double singleStep = pow(10.0, dim);
-            d->ui.clipX->setSingleStep(singleStep);
+            Private::ui.clipX->setSingleStep(singleStep);
         }
         {
             leny = leny / steps;
             int dim = static_cast<int>(log10(leny));
             double singleStep = pow(10.0, dim);
-            d->ui.clipY->setSingleStep(singleStep);
+            Private::ui.clipY->setSingleStep(singleStep);
         }
         {
             lenz = lenz / steps;
             int dim = static_cast<int>(log10(lenz));
             double singleStep = pow(10.0, dim);
-            d->ui.clipZ->setSingleStep(singleStep);
+            Private::ui.clipZ->setSingleStep(singleStep);
         }
 
         // set decimals
-        d->ui.clipView->setDecimals(minDecimals);
-        d->ui.clipX->setDecimals(minDecimals);
-        d->ui.clipY->setDecimals(minDecimals);
-        d->ui.clipZ->setDecimals(minDecimals);
+        Private::ui.clipView->setDecimals(minDecimals);
+        Private::ui.clipX->setDecimals(minDecimals);
+        Private::ui.clipY->setDecimals(minDecimals);
+        Private::ui.clipZ->setDecimals(minDecimals);
     }
 }
 
-Clipping* Clipping::makeDockWidget(Gui::View3DInventor* view)
+void Clipping::switchUi(Private* p)
```

Comment:
this method as a whole is hard to read due to not necessary empty lines that break reading flow. I also feel that it can be broken into few smaller ones with better names.